src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.VariableIdentifier.cs

```diff
@@ -29,11 +29,11 @@ public bool Exists
             }
             public override int GetHashCode()
             {
-                return Roslyn.Utilities.Hash.Combine(Symbol, ContainingSlot);
+                return Roslyn.Utilities.Hash.Combine(Symbol?.OriginalDefinition, ContainingSlot);
             }
             public bool Equals(VariableIdentifier other)
             {
-                return ((object)Symbol == null ? (object)other.Symbol == null : Symbol.Equals(other.Symbol)) && ContainingSlot == other.ContainingSlot;
+                return ((object)Symbol == null ? (object)other.Symbol == null : Symbol.OriginalDefinition.Equals(other.Symbol.OriginalDefinition)) && ContainingSlot == other.ContainingSlot;
```

Comment:
>Symbol.OriginalDefinition.Equals(other.Symbol.OriginalDefinition)) [](start = 80, length = 66)

I didn't understand this part of the change. What was the impact?

Reply:
> I didn't understand this part of the change. What was the impact?

The impact is that slight change in nested nullability of containing type (caused by SetPossiblyNullableReferenceTypeTypeParametersAsNullable, for example) won't have negative effect on allocating/finding slots. Container is already uniquely and accurately identified by ```ContainingSlot```, so we cant safely rely on definition to match members

---
In reply to: [230530406](https://github.com/dotnet/roslyn/pull/30913#discussion_r230530406) [](ancestors = 230530406)

Reply:
Thanks for the explanation

---
In reply to: [230954470](https://github.com/dotnet/roslyn/pull/30913#discussion_r230954470) [](ancestors = 230954470,230530406)